### PR TITLE
Docs (clean PR)

### PR DESCRIPTION
**Pull request by JordanCheney**
Ok @bklare @jklontz @sklum @caotto Lets move the discussion over here so we can have a clean(er) diff

### CODE REVIEW

**Comment by jklontz** [comment]
First pass through diff:
- Lots of empty `.md` files, can they be removed?
- Seems like docs/themes ought to be part of the source code for the website and not in this PR?
- Please provided the compiled docs on slack or elsewhere.

**Comment by JordanCheney** [comment]
Didn't see this comment sorry-

* I left the empty markdown files because I wanted a consistent structure for all of the classes (i.e each class has members, constructors, static functions, and functions). This doesn't have to be this way and is easy to change
* docs/themes should be moved
* Standby for html delivery

**Comment by jklontz** [comment]
I'd say remove the empty files, given the consistent structure, the absence of these files is probably more informative (and cleaner) than their presence as empty.

**Comment by jklontz** [comment]
Also, since markdown for the plugins is auto-generated from the python scripts, I would consider it a build artifact, and as such it should also be removed from the PR.

**Comment by JordanCheney** [comment]
Sounds good I will update

File: docs/docs/papers.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Related Publications
```
Comment (jklontz) [inline docs/docs/papers.md]: Remove this file?

File: docs/docs/tutorials.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+The command line isn't perfect for all situations however. So OpenBR exposes a [C++ API](api_docs/cpp_api.md) that can be embedded pretty much everywhere. Let's step through the example code at ```app/example/face_recognition.cpp``` and learn about using OpenBR as a library.
+
+Our main function starts with-
+
+		br::Context::initialize(argc, argv)
+
+This is the first step in any OpenBR application, it initializes the global context.
+
+	QSharedPointer<br::Transform> transform = br::Transform::fromAlgorithm("FaceRecognition");
```
Comment (jklontz) [inline docs/docs/tutorials.md]: Replace all uses of tabs with four spaces :)

File: docs/docs/api_docs/c_api/functions.md [hunk 3]
```diff
@@ -0,0 +1,1393 @@
...
+Checks if the provided algorithm is a classifier. Wrapper of [IsClassifier](../cpp_api/apifunctions.md#isclassifier).
+
+* **function definition:**
+
+		bool br_is_classifier(const char *algorithm)
+
+* **parameters:**
+
+	Parameter | Type | Description
+	--- | --- | ---
```
Comment (jklontz) [inline docs/docs/api_docs/c_api/functions.md]: I can't say I like unaligned style of table formatting. Thoughts?

**Comment by jklontz** [comment]
@sklum Can you review the doc content / style?

**Comment by sklum** [comment]
Happy to - it will have to be this weekend/next week though.

File: docs/docs/papers.md (hunk 1 above)
Comment (JordanCheney) [inline docs/docs/papers.md]: yep sorry missed this one

File: docs/docs/tutorials.md (hunk 2 above)
Comment (JordanCheney) [inline docs/docs/tutorials.md]: yeah :(

File: docs/docs/api_docs/c_api/functions.md (hunk 3 above)
Comment (JordanCheney) [inline docs/docs/api_docs/c_api/functions.md]: This style aligns better with the bulleted list in the rendered webpage IMO. 

* **parameters:**

Param | Type | Description
--- | --- | ---
thing | thing | thing

* **parameters:**

    Parameters | Type | Description
    --- | --- | ---
    thing | thing | thing

**Comment by jklontz** [comment]
Also, this PR should remove `share/openbr/Doxyfile.in` (not sure why there are edits to this file currently) and refactor `BR_BUILD_DOCUMENTATION` in `CMakeLists.txt`.

**Comment by JordanCheney** [comment]
I edited the doxyfile when I was trying to build the webpage directly from the doxygen output. These changes are now irrelevant and can be removed. I did add some custom aliases to our plugin documentation (br_paper for example to properly format papers). I can add these to the doxy file for completeness if you think it is a priority. I will also refactor cmake that's a good catch. One possible issue is that doxygen folds nicely into our cmake build while mkdocs does not (AFAIK). I will probably have to add a build_docs.sh script and remove the option completely from cmake. Thoughts on this?

**Comment by jklontz** [comment]
build_docs.sh is fine. no need to modify doxyfile before removal :)

File: docs/docs/index.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# OpenBR ![Overview](img/openbr_48x48.png)
+
+<p id=tagline>Open source, industry quality biometrics.</p>
```
Comment (sklum) [inline docs/docs/index.md]: A communal biometrics framework supporting the development of open algorithms and reproducible evaluations.

**Comment by sklum** [comment]
Alright @JordanCheney, I think I've reviewed all of the high level content.  Please let me know if there is a file you mentioned that I missed.  I standardized the installation docs, but I'd like to verify the installation steps for Windows and Raspian prior to the merge if you don't mind.  Also, a friendly reminder to fix the layout of a left column with respect to the repeated headings for the same category.

Can you briefly summarize what needs to be done for this to merge cleanly?  I'll start going through and filling in any of my `DOCUMENT ME` callouts in the meantime.

**Comment by sklum** [comment]
Alright, in terms of the things I wanted to accomplish, this can be merged.

A few notes:
* @JordanCheney will detail the process of hosting the docs on a ReadTheDocs server, which will continually build the documentation when a commit lands in master.  There will be a script, `download_docs.sh` to have documentation locally for those interested.
* After some discussion, we've decided that we would like to strictly adhere to the specifications laid out in `contribute.md`.  This involves documenting every property for every `Transform` such that users can understand all of the `Transform`s without looking at any source in similar fashion to OpenCV's documentation.  We can do this in rolling fashion, so no need to update all your `Transform`s at once.
* Installation instructions have been updated such that versions are standardized.  Linux, OSX, and Windows have been verified to work.  I'll do Raspian at a later date.

**Comment by sklum** [comment]
@JordanCheney FYI, training tutorial is reviewed/edited.

**Pull request merged**